src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/Utilities/EnvironmentHelper.cs

```diff
@@ -521,9 +536,27 @@ static byte ConvertFieldToByte (string fieldName, string envFile, int fileLine,
 			Assert.IsTrue (value.Length > 0, $"Field '{fieldName}' in {envFile}:{fileLine} is not a valid uint8_t value (not long enough)");
 
 			byte fv;
-			Assert.IsTrue (Byte.TryParse (value, out fv), $"Field '{fieldName}' in {envFile}:{fileLine} is not a valid uint8_t value (not a valid integer)");
+			Assert.IsTrue (TryParseInteger (value, out fv), $"Field '{fieldName}' in {envFile}:{fileLine} is not a valid uint8_t value (not a valid integer)");
 
 			return fv;
 		}
+
+		static bool TryParseInteger (string value, out uint fv)
+		{
+			if (value.StartsWith ("0x", StringComparison.Ordinal)) {
+				return UInt32.TryParse (value.Substring (2), NumberStyles.AllowHexSpecifier, null, out fv);
```

Comment:
Shouldn't this use `provider:CultureInfo.InvariantCulture` instead of `provider:null`? From the [`uint.Parse()` docs](https://docs.microsoft.com/en-us/dotnet/api/system.uint32.tryparse?view=net-6.0#system-uint32-tryparse(system-string-system-globalization-numberstyles-system-iformatprovider-system-uint32@)):

> If `provider` is `null`, the `NumberFormatInfo` object for the current culture is used.

Maybe this is more of a problem for floating-point values than integers, but I'm not sure we want a "current culture" integer parse here & elsewhere?

Reply:
The only problem can be parsing the thousands separator, but considering that this is a hexadecimal number that we're parsing, it shouldn't matter either. The only thing `InvariantCulture` may give us is, perhaps, slightly faster parsing.